### ISSUE TITLE
Exchange-based bindings need to be declared after Queue declarations

### ISSUE THREAD

**Issue by andrewgee** [ITV/bucky#13]
Currently bucky attempts to create bindings before the queues are declared causing it to fail. 

Stand-alone bindings are not affected: https://github.com/ITV/bucky/blob/9497c77e14c483b0347eb1a4daaa43d5e7f8f08a/core/src/main/scala/com/itv/bucky/decl/package.scala#L19

**Comment by BeniVF** [comment]
could you give me an example?

**Comment by andrewgee** [comment]
```
val testBadQueueName = QueueName("testbad")
val testBadQueue = Queue(testBadQueueName)

val testGoodQueueName = QueueName("testgood")
val testGoodQueue = Queue(testGoodQueueName)

val testExchangeName = ExchangeName("test")
val testExchange = Exchange(testExchangeName).binding(RoutingKey("testbad") -> testBadQueueName)
val testBinding = Binding(testExchangeName, testGoodQueueName, RoutingKey("testgood"), Map.empty)

val declarations = List(testExchange, testBinding, testBadQueue, testGoodQueue)```

If the `DeclarationLifecycle` is used for the `declarations` list, it could fail because the creation of the testbad binding could be attempted before the queue is created.

**Comment by BeniVF** [comment]
#14

**Comment by BeniVF** [comment]
It fixed on bucky version 1.0.0
